From 8dab79712f74d94bed6a9bd74fd61f225d92085c Mon Sep 17 00:00:00 2001
From: Anurag <anurag.gopinath@broadcom.com>
Date: Fri, 18 Jan 2013 23:39:11 +0530
Subject: [PATCH 665/761] SD/MMC : Fix the issue while reading MMC slot
 information from DTS file in LE mode

Based on Broadcom SDK 2.3.

Signed-off-by: Anurag <anurag.gopinath@broadcom.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/mmc/host/sdhci-xlp.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-xlp.c b/drivers/mmc/host/sdhci-xlp.c
index b510851..8feb9a5 100644
--- a/drivers/mmc/host/sdhci-xlp.c
+++ b/drivers/mmc/host/sdhci-xlp.c
@@ -63,8 +63,10 @@ static int findmaxslots(void) {
         int slots = 1;
         void *node;
         node = finddevice("/doms/dom@0/mmc");
-        if(node) 
+        if(node) {
             getprop(node, "slots", &slots, 4);
+            slots = be32_to_cpu(slots);
+        }
         return slots;    
 }
 
-- 
1.7.10.4

